### ISSUE TITLE
[Bug]: Set properties does not work properly in datacoord

### ISSUE THREAD

**Issue by xige-16** [milvus-io/milvus#34144]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master and 2.3
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. create collection A
2. collA.set_properties({"collection.ttl.seconds": 1000}), then datacoord.meta.collA.properties contain{"collection.ttl.seconds": "1000"}
3. Set another perproty, collA.set_properties({"collection.autocompaction.enabled": false}), then datacoord.meta.collA.properties  only contain collection.autocompaction.enabled, the old properties are overwritten

**Comment by yanliang567** [comment]
/assign @congqixia 
please help to reivew the prs as well

/unassign

**Comment by congqixia** [comment]
Could you please cherry pick fix into 2.4 branch? It's also needed, thx
/assign @xige-16

**Comment by xige-16** [comment]
> Could you please cherry pick fix into 2.4 branch? It's also needed, thx /assign @xige-16

Okey
